### ISSUE TITLE
Support pattern matching on arbitrary parameters for granule searches

### ISSUE THREAD

**Issue by betolink** [earthaccess-dev/earthaccess#198]
Based on some feedback from https://github.com/CryosphereVirtualLab/public-notebooks/issues/13 it would be great to add pattern matching to any parameter. With this enabled a user can search for patterns directly in any of the searchable fields i.e. the granule name: `MODGRNLD.*.daily.*` will filter only the daily granules for the MODGRNLD dataset, there are other use cases where this can be a handy thing to have and currently only the keyword parameter supports it out of the box.  To enable pattern matching we need to explicitly add it to the options parameter: `&options[readable_granule_name][pattern]=true` etc.

**Comment by jroebuck932** [comment]
I think this would be a great addition, this is a common workflow with our users.

**Comment by asteiker** [comment]
@briannapagan @chuckwondo Are you aware of whether this pattern matching search is supported in `python_cmr`? I think we'd need this supported in that library first before supporting it over in `earthaccess`?

**Comment by chuckwondo** [comment]
This specific issue is addressed very specifically: https://github.com/nsidc/earthaccess/blob/main/earthaccess/search.py#L630

More generally, however, `python_cmr` version 0.13.0 now includes an `option` method on the `Query` class: https://github.com/nasa/python_cmr/blob/develop/cmr/queries.py#L368

Note that there are no published docs for python_cmr (but there is an issue for it), but the docstring at the link above provides full usage detail for the `option` method.

Therefore, if we bump from 0.12.0 to 0.13.0, I think we should be able to do something like this (not tested), even though in the case of `granule_name`, this is unnecessary (see link at top of this comment):

```
earthaccess.search_data(
    ...,
    granule_name="MODGRNLD.*.daily.*",
    option=("readable_granule_name", "pattern", True),
)
```

However, that allows us to set only 1 option. If we want to set multiple options, we would need to directly make use of a query instance:

```
query = (
    earthaccess.DataGranules()
    .short_name(...)
    .granule_name("MODGRNLD.*.daily.*")
    .option("readable_granule_name", "pattern", True)
    .option(...)
    .option(...)
    ...
)
```

**Comment by asteiker** [comment]
Adding the decision label since this is related to the higher-level decision on incorporating all of python-cmr into `earthaccess`

**Comment by asteiker** [comment]
@chuckwondo Is this addressed with #1101 ?

**Comment by chuckwondo** [comment]
> [@chuckwondo](https://github.com/chuckwondo) Is this addressed with [#1101](https://github.com/nsidc/earthaccess/pull/1101) ?

No, that's a completely unrelated issue. This was addressed whenever python-cmr was bumped to 0.13.0, thus allowing the type of code I illustrated in the earlier comments.

**Comment by asteiker** [comment]
Great, I will close as addressed via updated python-cmr. Please re-open if I misinterpreted, or if there is still other work to implement on earthaccess.
